node/pkg/node/config_file_reader_test.go

```diff
@@ -0,0 +1,106 @@
+package node
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+)
+
+func NewTestRootCommand() *cobra.Command {
+	var ethRPC *string
+	var solRPC *string
+
+	// Define test configuration
+	testConfig := ConfigOptions{
+		FilePath:  "testdata",
+		FileName:  "test",
+		EnvPrefix: "TEST_GUARDIAN",
+	}
+
+	rootCmd := &cobra.Command{
+		Use:   "config_file_reader_test",
+		Short: "Unit test to test config file reader",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			// Initialize configuration using Viper
+			return InitFileConfig(cmd, testConfig) // Adjust the filename as needed
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			// Working with OutOrStdout/OutOrStderr allows us to unit test our command easier
+			out := cmd.OutOrStdout()
+
+			// Print the final resolved value from binding cobra flags and viper config
+			fmt.Fprintln(out, "ethRPC:", *ethRPC)
+			fmt.Fprintln(out, "solRPC:", *solRPC)
+		},
+	}
+
+	ethRPC = rootCmd.Flags().String("ethRPC", "", "Ethereum RPC URL")
+	solRPC = rootCmd.Flags().String("solRPC", "", "Solana RPC URL")
+
+	return rootCmd
+}
+
+func TestInitFileConfig(t *testing.T) {
+	// Set ethRPC with config file
+	// Tests that the config file is read and the default value is set
+	t.Run("config file", func(t *testing.T) {
```

Comment:
This is entirely fine, but is not generally idiomatic go. Generally speaking, you'll use `t.Run()` when you're doing [table driven tests](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests).

For this, it likely would be cleaner to make separate test functions instead of `TestInitFileConfig` that shares nothing.